src/diffusers/utils/export_utils.py

```diff
@@ -138,3 +140,354 @@ def export_to_video(
+    val_max: float = 1.0,
+    color_map: str = "Spectral",
+) -> Union[PIL.Image.Image, List[PIL.Image.Image]]:
+    """
+    Visualizes depth maps, such as predictions of the `MarigoldDepthPipeline`.
+
+    Args:
+        depth (`Union[PIL.Image.Image, np.ndarray, torch.FloatTensor, List[PIL.Image.Image], List[np.ndarray],
+            List[torch.FloatTensor]]`): Depth maps.
+        val_min (`float`, *optional*, defaults to `0.0`): Minimum value of the visualized depth range.
+        val_max (`float`, *optional*, defaults to `0.0`): Maximum value of the visualized depth range.
+        color_map (`str`, *optional*, defaults to `"Spectral"`): Color map used to convert a single-channel
+                  depth prediction into colored representation.
+
+    Returns: `PIL.Image.Image` or `List[PIL.Image.Image]` with depth maps visualization.
+    """
+    if val_max <= val_min:
+        raise ValueError(f"Invalid values range: [{val_min}, {val_max}].")
+
+    def visualize_depth_one(img, idx=None):
+        prefix = "Depth" + (f"[{idx}]" if idx else "")
+        if isinstance(img, PIL.Image.Image):
+            if img.mode != "I;16":
+                raise ValueError(f"{prefix}: invalid PIL mode={img.mode}.")
+            img = np.array(img).astype(np.float32) / (2**16 - 1)
+        if isinstance(img, np.ndarray) or torch.is_tensor(img):
+            if img.ndim != 2:
+                raise ValueError(f"{prefix}: unexpected shape={img.shape}.")
+            if isinstance(img, np.ndarray):
+                img = torch.from_numpy(img)
+            if not torch.is_floating_point(img):
+                raise ValueError(f"{prefix}: unexected dtype={img.dtype}.")
+        else:
+            raise ValueError(f"{prefix}: unexpected type={type(img)}.")
+        if val_min != 0.0 or val_max != 1.0:
+            img = (img - val_min) / (val_max - val_min)
+        img = colormap(img, cmap=color_map, bytes=True)  # [H,W,3]
+        img = PIL.Image.fromarray(img.cpu().numpy())
+        return img
+
+    if depth is None or isinstance(depth, list) and any(o is None for o in depth):
+        raise ValueError("Input depth is `None`")
+    if isinstance(depth, np.ndarray) or torch.is_tensor(depth):
+        if depth.ndim == 2:
+            return visualize_depth_one(depth)
+        if depth.ndim == 3:
+            return [visualize_depth_one(img, idx) for idx, img in enumerate(depth)]
+        if depth.ndim == 4:
+            if depth.shape[1] != 1:
+                raise ValueError(f"Unexpected input shape={depth.shape}, expecting [N,1,H,W].")
+            return [visualize_depth_one(img[0], idx) for idx, img in enumerate(depth)]
+        raise ValueError(f"Unexpected input shape={depth.shape}: expecting 2, 3, or 4 dimensions.")
+    elif isinstance(depth, list):
+        return [visualize_depth_one(img, idx) for idx, img in enumerate(depth)]
+    else:
+        raise ValueError(f"Unexpected input type: {type(depth)}")
+
+
+def export_depth_to_16bit_png(
```

Comment:
these functions export just images (instead of different file formats), no? I wonder if it makes sense to move the marigold folder and make them importable from there - 

cc @sayakpaul what do you think?

Reply:
ohh actually I think these methods should be part of MarigoldImageProcessor!

Reply:
like additional util methods we provide to users that they can use outside of the pipeline 
we did similar with ip-adapter mask processing https://huggingface.co/docs/diffusers/using-diffusers/ip_adapter#ip-adapter-masking

Reply:
I think they should be added to Marigold image processor